### PR TITLE
Fix downloads

### PR DESCRIPTION
**Pull request by bb4242**
It seems that downloads have been broken for a little while (I've received a couple of emails about it recently).  Investigating, it seems that the media URL we were previously using has changed a bit.  This PR follows redirects from the actual audio URL we get from dharmaseed, which should hopefully be a more reliable way to download talks.

### CODE REVIEW

**Comment by bb4242** [comment]
Also, I'm not sure exactly what changed, but I couldn't get the downloaded talks to play correctly anymore when using external storage (MediaPlayer was complaining about not being able to load the file).  So, I changed the download location to be in the app's internal storage location instead.

File: app/src/main/java/org/dharmaseed/android/TalkRepository.java [hunk 1]
```diff
@@ -185,6 +187,36 @@ public Cursor getTalksByCenter(
                 DBManager.C.Center.TABLE_NAME + "." + DBManager.C.Center.ID + "=" + venueId);
     }
 
+    /** Check if we have talks downloaded in old/inaccessible locations, and mark them as not downloaded if so
```
Comment (jakewilson) [inline app/src/main/java/org/dharmaseed/android/TalkRepository.java]: can you put this comment on the next line?

File: app/src/main/java/org/dharmaseed/android/TalkRepository.java [hunk 2]
```diff
@@ -185,6 +187,36 @@ public Cursor getTalksByCenter(
...
+                    DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.ID)));
+            String path = downloaded.getString(downloaded.getColumnIndexOrThrow(DBManager.getAlias(
+                    DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.FILE_PATH))).trim();
+
+            String fullPath = new File(path).getAbsolutePath();
+            String downloadPath = TalkManager.getDir(dbManager.getContext()).getAbsolutePath();
+
+            if (! fullPath.startsWith(downloadPath)) {
+                Log.w(LOG_TAG, "Removing old download for talk " + id);
+                dbManager.removeDownload(id);
```
Comment (jakewilson) [inline app/src/main/java/org/dharmaseed/android/TalkRepository.java]: this removes our log for it in the DB, but it leaves the file on their system doesn't it? Can we also delete the file from the system? Or, even better, move the file from the old dir to the new one?

I would hate for people to suddenly lose all of their downloaded talks with a new update, although it seems like they haven't been able to play them for a little while anyway. What are your thoughts?

**Comment by jakewilson** [comment]
This looks awesome! Just have a little concern, as mentioned above.

**Comment by jakewilson** [comment]
Also this looks like it would solve #43

File: app/src/main/java/org/dharmaseed/android/TalkRepository.java (hunk 1 above)
Comment (bb4242) [reply app/src/main/java/org/dharmaseed/android/TalkRepository.java]: Done!

File: app/src/main/java/org/dharmaseed/android/TalkRepository.java (hunk 2 above)
Comment (bb4242) [reply app/src/main/java/org/dharmaseed/android/TalkRepository.java]: Yes, that's a great idea, thanks!  I've updated it to attempt to move the file to the new location, and if that succeeds, to re-add it to the downloads table.  I've tested this to verify that it does indeed preserve previously downloaded files when upgrading the app.

**Comment by bb4242** [comment]
> Also this looks like it would solve #43

Agreed, I think we can close that one once this merges.

**Comment by jakewilson** [comment]
Looks good to me. Thanks for adding this!

**Pull request merged**